src/v2/BaseLoginRouter.js

```diff
@@ -103,7 +103,10 @@ export default Router.extend({
     //    -> refresh page
     //    -> introspect using options.stateHandle
     if (lastResponse) {
-      sessionStorageHelper.setStateHandle(idxResponse?.context?.stateHandle);
+      const stateHandle = idxResponse?.context?.stateHandle;
+      const appId = idxResponse?.context?.app?.value?.id;
+      // store stateHandle per app context
+      sessionStorageHelper.setStateHandle(stateHandle, appId);
```

Comment:
Assuming appId will exist for sign into app flows

Reply:
Assuming that `idxResponse` aka `this.appState.get('idx')` will change when signing into a different app